src/Microsoft.Health.Dicom.Core/Features/Workitem/WorkitemQueryTagService.cs

```diff
@@ -53,9 +63,24 @@ public async Task<IReadOnlyCollection<WorkitemQueryTagStoreEntry>> GetQueryTagsA
             return await _queryTagCache.GetAsync(cancellationToken: cancellationToken);
         }
 
-        private async Task<IReadOnlyCollection<WorkitemQueryTagStoreEntry>> ResolveQueryTagsAsync(CancellationToken cancellationToken)
+        private async Task<IReadOnlyCollection<QueryTag>> ResolveQueryTagsAsync(CancellationToken cancellationToken)
         {
-            return await _indexWorkitemStore.GetWorkitemQueryTagsAsync(cancellationToken);
+            var workitemQueryTags = await _indexWorkitemStore.GetWorkitemQueryTagsAsync(cancellationToken);
+
+            foreach (var tag in workitemQueryTags)
+            {
+                if (_dicomTagParser.TryParse(tag.Path, out DicomTag[] dicomTags))
+                {
+                    tag.PathTags = Array.AsReadOnly(dicomTags);
```

Comment:
Is this necessary? Surely no one is casting the `IReadOnlyCollection<T>` to a mutable type internally

Reply:
We couldnt assign dicomTags directly to PathTags, so we have create a readOnly list